libraries/shared/watcher/storage_watcher.go

```diff
@@ -31,22 +31,18 @@ import (
 	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
 )
 
-type StorageWatcher struct {
-	db             *postgres.DB
-	StorageFetcher fetcher.IStorageFetcher
-	Queue          storage.IStorageQueue
-	Transformers   map[common.Address]transformer.StorageTransformer
+type IStorageWatcher interface {
+	AddTransformers(initializers []transformer.StorageTransformerInitializer)
+	Execute(diffsChan chan utils.StorageDiff, errsChan chan error, queueRecheckInterval time.Duration)
 }
 
-func NewStorageWatcher(fetcher fetcher.IStorageFetcher, db *postgres.DB) StorageWatcher {
-	transformers := make(map[common.Address]transformer.StorageTransformer)
-	queue := storage.NewStorageQueue(db)
-	return StorageWatcher{
-		db:             db,
-		StorageFetcher: fetcher,
-		Queue:          queue,
-		Transformers:   transformers,
-	}
+type StorageWatcher struct {
```

Comment:
Stupid question, but why are there 3 storage watchers (CSV, geth, and this one)?

Reply:
`StorageWatcher` is like the base or parent struct that `GethStorageWatcher` and `CsvStorageWatcher` inherit from since there is a lot of shared behavior. I'd be happy to rethink, or update this approach if this is confusing, or not ideal.